src/test/kotlin/com/igorwojda/integer/factorial/challenge.kt

```diff
@@ -4,7 +4,11 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun factorial(n: Int): Int {
-    TODO("not implemented")
+    var result=1
```

Comment:
Re-write this:
1. Using a folding operation: https://kotlinlang.org/docs/collection-aggregate.html#fold-and-reduce
2. Using a recursive algorithm
Please compare the running time of the two solutions.
If you're curious, look at tail recursive functions (https://kotlinlang.org/docs/reference/functions.html#tail-recursive-functions). See if you can write factorial as a tailrec function.